_kbarticlespages/repository-best-practices.md

```diff
@@ -0,0 +1,67 @@
+excerpt: Pages guide for mainitning GitHub hosted repositories
+---
+
+Adhering to a set of best practices when it comes to managing GitHub repositories can greatly ease the maintenance burden and drastically improve the developer experience for websites hosted on cloud.gov Pages. It must be stated that this document should serve as a constructive conversation piece for agencies and not as an audit mechanism. Although by utilizing certain code scanning tools agencies may find it easier to identify and cover certain security controls within their respective Customer Responsibility Matrix (CRM). Here is a useful guide for you to get started with implementing some best practices regarding user permissions and security for your GitHub hosted repositories.
+
+## **Public vs Private repositories**
+
+A repository does *not need* to be public in order for Pages to be able to build the website, both public and private repositories are acceptable. When it comes to Pages there are certainly some unique considerations for each.
+
+ #### **Public**
+
+##### **Pros**
+  - Easy for Pages engineers to access the codebase and provide support/debug
+  - Transparency/High visibility
+  - Able to utilize CodeQL free of charge
+  - Seamless OpenSSF Allstar integration
+  - Collaboration with outside contributors
+  - Compliance with open source and transparency initiatives per agency/   program 
+
+##### **Cons**
+  - Any sensitive information inadvertently hardcoded is immediately visible 
+
+#### **Private**
+
+##### **Pros**
+  - Restrict access to authorized collaborators and contributors (for any support issues a Pages engineer will request to be added as a collaborator to the repository if necessary)
+  - Risk mitigation
+  - Test/Make changes to a website without exposing commits to the public
+  - Safeguard against any potential leaks of sensitive information
+
+##### **Cons**
+  - Not able to utilize CodeQL for free
+  - Potentially out of compliance with any agency/program transparency initiative
+
+## **User Permissions**
+
+It is important to designate admin/owner access of the repository to an individual(s) who is either the persistent site owner or a federal employee with the “manager” role within the Pages organization. It is strongly recommended to not grant admin/owner privileges of the repository to any contractors or other temporary employees. This is such due to certain instances where actions will need to be taken by either Pages engineers or customers where it will be necessary to have an active user with admin access to a website's repository. Additionally when adding contributors/collaborators to a private repository it should mirror the member count that your organization has within Pages. Please note that removing an organization member from an organization within Pages will **not** automatically remove them from the repository within GitHub. They will still retain whatever read/write access they have in GitHub until explicitly removed.
```

Comment:
```suggestion
It is important to designate admin/owner access of the repository to one or more individuals who are either the persistent site owners or else federal employees with the “manager” role within the site's Pages organization. It is strongly recommended to not grant admin/owner privileges of the repository to any contractors or other temporary employees. This is because in certain instances actions may need to be taken by either Pages engineers or by customers that require an active user with admin access to a website's repository. Additionally, when adding contributors or collaborators to a private repository it should mirror the member count that your organization has within Pages. Please note that removing an organization member from an organization within Pages will **not** automatically remove them from the repository within GitHub. They will still retain whatever read/write access they have in GitHub until explicitly removed.
```